File: backend/app/schemas/file.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class FileItem(BaseModel):
    name: str
    path: str
    is_directory: bool
    size: Optional[int] = None
    modified_time: datetime
    file_type: Optional[str] = None
    permissions: str


class DirectoryListing(BaseModel):
    current_path: str
    parent_path: Optional[str] = None
    items: List[FileItem]
    total_items: int


class RenameRequest(BaseModel):
    new_name: str


class MP3TagData(BaseModel):
    title: Optional[str] = None
    artist: Optional[str] = None
    album: Optional[str] = None
    genre: Optional[str] = None
    year: Optional[str] = None
    tracknumber: Optional[str] = None


class MP3TagUpdate(BaseModel):
    title: Optional[str] = None
    artist: Optional[str] = None
    album: Optional[str] = None
    genre: Optional[str] = None
    year: Optional[str] = None
    tracknumber: Optional[str] = None

File: frontend/src/types/file.ts
export interface FileItem {
  name: string;
  path: string;
  isDirectory: boolean;
  size?: number;
  modifiedTime: string;
  fileType?: string;
  permissions: string;
}

export interface DirectoryListing {
  currentPath: string;
  parentPath?: string;
  items: FileItem[];
  totalItems: number;
}

export interface RenameRequest {
  newName: string;
}

export interface MP3TagData {
  title?: string;
  artist?: string;
  album?: string;
  genre?: string;
  year?: string;
  tracknumber?: string;
}

export interface MP3TagUpdate {
  title?: string;
  artist?: string;
  album?: string;
  genre?: string;
  year?: string;
  tracknumber?: string;
}
